tests/sles4sap/desktop_icons.pm

```diff
@@ -0,0 +1,36 @@
+# SUSE's openQA tests
+#
+# Copyright © 2018 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: SLES for SAP Applications default desktop icons check
+# Maintainer: Alvaro Carvajal <acarvajal@suse.de>
+
+use base "sles4sap";
+use testapi;
+use strict;
+
+sub run {
+    my ($self) = @_;
+
+    select_console 'x11';
+    # Check for SLES4SAP Desktop icons
+    if (check_screen 'sles4sap-desktop') {
```

Comment:
Please, try to use aseert_screen on multiple tags and then do your logic using match_has_tag function. Then we avoid waiting when check_screen times out.

Reply:
The first version I wrote for this used a call to `assert_screen [qw(sles4sap-desktop generic-desktop)]` and then a call to `match_has_tag` as the `if` condition instead of the `check_screen`; but basically rewrote it because `generic-desktop` matches the SLES4SAP desktop (100% match as well) with and without the presence of the icons, and didn't want to record a soft failure when the desktop icons are present but generic-desktop is matched. I need to call a `record_soft_fail` when the screen matches `generic-desktop` and doesn't match `sles4sap-desktop` which is why I went with `check_screen` in this case. Perhaps I need to add a `no_wait => 1`?

Reply:
Hmm, that's weird because my understanding is that assert_screen does matching in the sequence parameters are given and as long as 'sles4sap-desktop' doesn't have 'generic-desktop' tag it should work. @okurz am I missing anything here?
Otherwise we need some timeout for check_screen as screen won't appear immediately.

Reply:
I tested like this

```
    assert_screen [qw(sles4sap-desktop generic-desktop)];
    if (match_has_tag 'sles4sap-desktop') {
```

And, as you can see in http://mango.suse.de/tests/201#step/desktop_icons/4, it's matching first to `generic-desktop` and reporting a soft fail on 12SP3 when the icons are there.

Reply:
Good to know, thanks! So then let's leave it as it is. As a side note, if you set WORKAROUND property on a needle it will have higher priority than normal needles, which could help in this scenario if we would already have code in place. As an alternative we could create a needle which has icons area included.
Anyway, please see my comment in needles MR and we can give it a shot on production.